src/Interpreters/DDLWorker.h

```diff
@@ -201,6 +214,8 @@ class DDLWorker
 
     const CurrentMetrics::Metric * max_entry_metric;
     const CurrentMetrics::Metric * max_pushed_entry_metric;
+
+    std::unordered_map<String, std::pair<ZooKeeperPtr, zkutil::EphemeralNodeHolderPtr>> active_node_holders;
```

Comment:
Maybe we can use `host_fqdn_id` and create one active node per replica, but I'm not sure

Reply:
Multiple host_id can point to the same instance. For example, both `domain1-host2.aws.com` and `domain2-host2.aws.com` are resolved to `host2`. 

So my idea is that for each host_id in each cluster, the DDL worker creates a `host_id` dir in `replicas_dir` in `initializeReplication`. In the main thread, each DDL worker checks if the `host_id` under `replicas_dir` belongs to itself, it will create an `active` node under the `host_id` dir to indicate that the `host_id` is running.

Reply:
Because multiple host_ids can belong to 1 instance, I use a map to hold the ephemeral nodes.

Reply:
Ok, it makes sense. Without creating multiple active nodes we would have to perform complex checks (like in `findCurrentHostID`) to check if a host is active or not